packages/amplify_native_legacy_wrapper/android/src/main/kotlin/com/amazonaws/amplify/amplify_native_legacy_wrapper/LegacyNativePluginPigeon.java

```diff
@@ -23,38 +23,64 @@
 import java.util.Map;
 
 /** Generated class from Pigeon. */
-@SuppressWarnings({"unused", "unchecked", "CodeBlock2Expr", "RedundantSuppression"})
+@SuppressWarnings({"unused", "unchecked", "CodeBlock2Expr", "RedundantSuppression", "serial"})
 public class LegacyNativePluginPigeon {
+
+  /** Error class for passing custom error details to Flutter via a thrown PlatformException. */
+  public static class FlutterError extends RuntimeException {
+
+    /** The error code. */
+    public final String code;
+
+    /** The error details. Must be a datatype supported by the api codec. */
+    public final Object details;
+
+    public FlutterError(@NonNull String code, @Nullable String message, @Nullable Object details) 
+    {
+      super(message);
+      this.code = code;
+      this.details = details;
+    }
+  }
+
   @NonNull
-  private static ArrayList<Object> wrapError(@NonNull Throwable exception) {
+  protected static ArrayList<Object> wrapError(@NonNull Throwable exception) {
     ArrayList<Object> errorList = new ArrayList<Object>(3);
-    errorList.add(exception.toString());
-    errorList.add(exception.getClass().getSimpleName());
-    errorList.add(
-      "Cause: " + exception.getCause() + ", Stacktrace: " + Log.getStackTraceString(exception));
+    if (exception instanceof FlutterError) {
+      FlutterError error = (FlutterError) exception;
+      errorList.add(error.code);
+      errorList.add(error.getMessage());
+      errorList.add(error.details);
+    } else {
+      errorList.add(exception.toString());
+      errorList.add(exception.getClass().getSimpleName());
+      errorList.add(
+        "Cause: " + exception.getCause() + ", Stacktrace: " + Log.getStackTraceString(exception));
+    }
     return errorList;
   }
 
   public interface Result<T> {
+    @SuppressWarnings("UnknownNullness")
     void success(T result);
 
-    void error(Throwable error);
+    void error(@NonNull Throwable error);
```

Comment:
Question - why do we remove @NonNull in Kotlin yet the latest Pigeon is adding @NonNull to Java code?

Reply:
`@NonNull` is only relevant in Java where the type system has no concept of nullability. Adding them in Kotlin is like adding them in Dart... doesn't add much.